paper-server/patches/sources/net/minecraft/server/MinecraftServer.java.patch

```diff
@@ -769,6 +769,51 @@
++                        final String repo = "PaperMC/Paper";
++                        int distance = com.destroystokyo.paper.PaperVersionFetcher.DISTANCE_ERROR;
++
++                        if (build.buildNumber().isEmpty() && build.gitCommit().isEmpty()) {
++                            LOGGER.warn("*** You are running a development version without access to version information ***");
++                        } else {
++                            if (buildNumber.isPresent()) {
++                                distance = com.destroystokyo.paper.PaperVersionFetcher.fetchDistanceFromSiteApi(build, buildNumber.getAsInt());
++                            } else {
++                                final Optional<String> gitBranch = build.gitBranch();
++                                final Optional<String> gitCommit = build.gitCommit();
++                                if (gitBranch.isPresent() && gitCommit.isPresent()) {
++                                    distance = com.destroystokyo.paper.PaperVersionFetcher.fetchDistanceFromGitHub(repo, gitBranch.get(), gitCommit.get());
++                                    }
++                                }
++
++                            switch (distance) {
++                                case com.destroystokyo.paper.PaperVersionFetcher.DISTANCE_ERROR -> LOGGER.error("*** Error obtaining version information! Can't fetch version info ***");
++                                case 0 -> {
++                                    COMPONENT_LOGGER.info(net.kyori.adventure.text.Component.text("*** You can safely disregard the previous warning ***", net.kyori.adventure.text.format.NamedTextColor.GREEN));
++                                    COMPONENT_LOGGER.info(net.kyori.adventure.text.Component.text("*** You're running the latest version! ***", net.kyori.adventure.text.format.NamedTextColor.GREEN));
++                                }
++                                case com.destroystokyo.paper.PaperVersionFetcher.DISTANCE_UNKNOWN -> LOGGER.warn("*** You are running an unknown version! Can't fetch version info ***");
++                                case 5 -> {
++                                    LOGGER.error("*** You are " + distance + " builds behind!");
++                                    LOGGER.error("*** Please download a new build from https://papermc.io/downloads/paper ***");
++                                }
++                                default -> {
++                                    LOGGER.warn("*** Currently you are " + distance + " builds behind");
++                                    LOGGER.warn("*** It's highly recommended to download a new build from https://papermc.io/downloads/paper ***");
++                                }
++                            }
++                        };
++                    }
++            } catch (Throwable t) {
++                t.printStackTrace();
++            }
++        });
++            // Paper end - improve outdated version check
```

Comment:
i feel this can be moved to a method in PaperVersionFetcher because you use things like urls already defined in that class

Reply:
Yeah you're right will move this

Reply:
moved